### ISSUE TITLE
Generate Random Objects!

### ISSUE THREAD

**Issue by nvasilakis** [victorquinn/chancejs#194]
Hi -- thanks for the great library!

It seems it would be extremely easy and useful to implement getting whole random objects -- like person, web etc. It would mostly compile a set of random values for their fields.

This would be very useful for testing (suppose somebody wants randomly generated objects). A similar, albeit more advanced, idea would be to have a `chance.newObject(obj)` where obj is an object describing the _structure_ of the resulting object (i.e., allowing chance to fill in the values of the template).

I would be interested to hear thoughts, and if you agree on the features I can send a pull request:)

Thanks again!

**Comment by oliversalzburg** [comment]
Sounds like a duplicate of #82

**Comment by nvasilakis** [comment]
Not sure -- I mean yes, at a high level it is a duplicate, bit I am proposing a very specific subset that I think is worth implementing. chancejs is all about simplifying getting arbitrary values of entities that are used commonly (_e.g., Person, Address, etc_) so why not allow getting ready-to-consume high level objects (the same very person, address etc..).

**Comment by oliversalzburg** [comment]
The problem I see is, the person object *you* require might differ from the person object someone else might need. Once you need something complex, it's more useful to have proper, small building blocks so that you can *build* your complex object. Having the complex object served to you might not meet your specific requirements.

**Comment by franciscop** [comment]
@oliversalzburg can you please clarify `the person object you require might differ from the person object someone else might need`? Do you mean different in content or structure? (sorry I'm not native English speaker so that might have made me need the extra explanation)

@nvasilakis please [check my pending pull request](https://github.com/victorquinn/chancejs/pull/192) for something similar I suggested that can be obtained with that pull:

```js
// Generate an array with 20 random numbers
chance.array({ length: 20, fill: function(){
  return {
    firstname: chance.first(),
    lastname: chance.last(),
    age: chance.age(),
    image: chance.svg()
    // ...
    }
  }
});
```

However after creating it I also thought of a way of creating objects exactly for the same need that you have. The properties (keys) of the object would be injected, then the values might be random ints or generated using a request object like this:

```js
// Just generating random ints (maybe not so useful):
chance.object({ keys: ['firstname', 'lastname', 'age'] });

// Generate an object with random values generated by the specified chancejs function:
chance.object({ keys: { firstname: 'first', lastname: 'last', age: 'age' } });
// Alternative (injecting a callback):
chance.object({ keys: { firstname: chance.first, lastname: chance.last, age: chance.age } });
```

A simpler alternative (but not extensible) would be to just pass the object as you said:

```js
chance.object({ firstname: 'first', lastname: 'last', age: 'age' });
chance.object({ firstname: chance.first, lastname: chance.last, age: chance.age });
```

**Comment by oliversalzburg** [comment]
@franciscop I actually mean both.

**Comment by caimen** [comment]
@oliversalzburg I tend to agree with you, generating random objects with so many possible different permutations and requirements would I think be an endless rabbit hole.  It's a rabbit hole I've been down before and there is no end in sight.  Right now chance js gives you the ability to generate all sort of of data types.  Users can use those data types to create there own random objects very easily.  I don't see the advantage of having the library attempt do things like that for you and in the end seems like a futile endeavor.

**Comment by joshlangner** [comment]
`I don't see the advantage of having the library attempt do things like that for you...`

... But isn't that the point of a Library? Being able to do things for you :)

I came across this issue while seeing whether ChanceJS could randomly drop in some bad data to sets of good data. Let's say I wanted a list of Names:

```Names = ["john", "sarah", "gene"];```

Now I'd like for Chance, when it generates it, to also drop in some bad data:

```Names = ["john", undefined, "gene", 42];```

This would be super-useful for testing-purposes, to be able to mock data that contains both good and bad attributes. If as part of chance's generation functions we could pass in a param such as `dirty` as well as potentially a function defining the `type` of `dirty`, e.g. wrong types, falsey or undefineds, etc. ... that would definitely be useful, and as I referenced earlier, it would enable the Library to save me (the developer) a lot of time.

**Comment by ganqqwerty** [comment]
What about generating objects by their jsonschema or typescript type or class descriptions?
